apps/api/src/models/post.ts

```diff
@@ -0,0 +1,29 @@
+import mongoose, { Schema } from 'mongoose';
+import Post from '../interfaces/post';
+
+export const postSchema = new Schema<Post>(
+  {
+    title: {
+      type: String,
+      required: [true, 'Property is required']
+    },
+    img: {
+      type: String
+    },
+    description: {
+      type: String,
+      required: [true, 'Property is required']
+    },
+    category: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'Category'
+    },
+    comments: {
+      type: [{ content: String, author: String }]
```

Comment:
This is a correct approach to have documents nested into another document, to have a better performant we can implement a reference to a different document. Something like this: 

```
comments: [{
  type: Schema.Types.ObjectId,
  ref: 'Comment' // You will need to create a comment model
}]
```